### PR TITLE
Homework D04 sberthely

### PR DESCRIPTION
**Pull request by sberthely**
Homework D04 sberthely

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,19 +11,34 @@
...
+import random
 
 # Body
 
+def random_guess():
+	num_random = random.randint(1, 25)
+	print(num_random)
+	number_tries = 0
+
+	while(number_tries < 5):
```
Comment (danielsgriffin) [inline D04_ex00.py]: Doesn't have to be in parenthesis.

File: D04_ex00.py [hunk 2]
```diff
@@ -11,19 +11,34 @@
...
+		number_tries += 1
 
+		user_number = int(input('Pick a number between 1 - 25: '))
```
Comment (danielsgriffin) [inline D04_ex00.py]: We wanted to use a `try`/`except` here to help with the validation

File: HW04_ch07_ex02.py [hunk 3]
```diff
@@ -25,11 +25,18 @@
 
 # Body
 
+def eval_loop():
+
+	while True:
+		s = input('Write anything: ')
+		if s == 'done':
+			break
+		print(eval(s))
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "continue until the user enters 'done', and then return the value of the last expression it evaluated"

File: HW04_ch08_ex00.py [hunk 4]
```diff
@@ -21,14 +21,22 @@
 
 # Body
 
+def count(s, letter):
+	word = 'banana'
+	count = 0
+	for letter in word:
+		if letter == 'a':
+			count = count + 1
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: count += 1

File: HW04_ch08_ex04.py [hunk 5]
```diff
@@ -34,25 +34,27 @@ def any_lowercase2(s):
 
 
 def any_lowercase3(s):
-    """Explain what is wrong, if anything, here.
-    """
+    # Explain what is wrong, if anything, here.
+    # What is wrong is that it only evaluates the last letter.
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: Well, it does evaluate all letters. It returns based off the value of only the last letter.

File: HW04_ch08_ex04.py [hunk 6]
```diff
@@ -34,25 +34,27 @@ def any_lowercase2(s):
...
         flag = c.islower()
     return flag
 
 
 def any_lowercase4(s):
-    """Explain what is wrong, if anything, here.
-    """
+    #Explain what is wrong, if anything, here.
+    # What is wrong is that in the first execution the variable 'flag' can get the value 'True' and after that it will always returns 'True' no matter what 
+    # happens with the sentence 'c.islower()'. It will only work partially if the fisrt letter is capital.
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: I'm confused. What is the goal of this function? Can you show me an example of this function failing?

File: HW04_ch08_ex04.py [hunk 7]
```diff
@@ -34,25 +34,27 @@ def any_lowercase2(s):
...
         flag = flag or c.islower()
+        #print(c + ' ' + str(flag))
     return flag
 
 
 def any_lowercase5(s):
-    """Explain what is wrong, if anything, here.
-    """
+    # Explain what is wrong, if anything, here.
+    # This function works fine for me, the function evaluates each letter and quits the the execution as soon as it finds the first capital letter.
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: Again, what is the goal?

File: HW04_ch08_ex05.py [hunk 8]
```diff
@@ -2,3 +2,21 @@
...
+	for letter in s:
+		new_string = new_string + chr(ord(letter) + i)
+	return new_string
+
+#ord #converts a character to a numeric code ('a' - ord = 97 and there are 32 letters alphabet)
+
+#chr #which converts numeric codes to characters.
+
+def main():
+	s = input('Write a phrase: ')
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Taking the input like this was not required

File: HW04_ch08_ex05.py [hunk 9]
```diff
@@ -2,3 +2,21 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: Did you test your code?
```
Book Examples:
 OK  'A'     by 3   => got: 'D'     expected: 'D'
  X  'Z'     by 1   => got: '['     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
  X  'melon' by -10 => got: 'c[bed' expected: 'cubed'
 OK  'IBM'   by -1  => got: 'HAL'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
  X  'xyz'   by 1   => got: 'yz{'   expected: 'yza'
 OK  'ABC'   by 1   => got: 'BCD'   expected: 'BCD'
  X  'XYZ'   by 1   => got: 'YZ['   expected: 'YZA'
  X  'Az'    by 1   => got: 'B{'    expected: 'Ba'
  X  'Aa'    by 26  => got: '[{'    expected: 'Aa'
  X  'abc'   by -1  => got: '`ab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: '@AB'   expected: 'ZAB'
 OK  'XYZ'   by -1  => got: 'WXY'   expected: 'WXY'
  X  'Az'    by -1  => got: '@y'    expected: 'Zy'
  X  'Aa'    by -26 => got: "'G"    expected: 'Aa'
```

**Comment by danielsgriffin** [comment]
Good job. Make sure you test your code?

File: HW04_string2.py [hunk 10]
```diff
@@ -42,8 +53,23 @@ def not_bad(s):
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
-    # +++your code here+++
-    return
```
Comment (lizlee0225) [inline HW04_string2.py]: I think this might exit the function right away.

File: D04_ex00.py (hunk 2 above)
Comment (lizlee0225) [inline D04_ex00.py]: User might put something other than a number.

File: HW04_ch08_ex00.py [hunk 11]
```diff
@@ -21,14 +21,22 @@
 
 # Body
 
+def count(s, letter):
+	word = 'banana'
+	count = 0
+	for letter in word:
+		if letter == 'a':
```
Comment (lizlee0225) [inline HW04_ch08_ex00.py]: Word and letter should be s and letter from the argument

File: HW04_string2.py [hunk 12]
```diff
@@ -17,7 +17,13 @@
 # If the string length is less than 3, leave it unchanged.
 # Return the resulting string.
 def verbing(s):
-    # +++your code here+++
+    if len(s) < 3:
+        return s
+    elif len(s) >= 3:
+        if s.find('ing', len(s) - 3, len(s)) >= 0:
```
Comment (lizlee0225) [inline HW04_string2.py]: Didn't know you could use 3 perimeters with find function. Cool!

**Pull request closed without merging**